core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -216,26 +219,23 @@ class DruidBeamMaker[A: Timestamper](
 
 object DruidBeamMaker
 {
-  def generateBaseFirehoseId(dataSource: String, segmentGranularity: Granularity, ts: DateTime, partition: Int) = {
+  def generateBaseFirehoseId(dataSource: String,taskDuration:Period, ts: DateTime, partition: Int) = {
     // Not only is this a nasty hack, it also only works if the RT task hands things off in a timely manner. We'd rather
     // use UUIDs, but this creates a ton of clutter in service discovery.
+    //It may also break if task duration is changed across RT tasks
```

Comment:
Lets say earlier duration is less than an hour than the firehose id will be "datasource-<minute of hour>-partition" and later if application is restarted and task duration is changed to more than an hour than tranquility won't be able to identify the older running task as it would be looking for "datasource-<hour ofday>-<minute of hour>-partition"